keyboards/grid/grid_650/grid_650.c

```diff
@@ -0,0 +1,616 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#define wdt_intr_enable_copy(value)   \
+__asm__ __volatile__ (  \
+    "in __tmp_reg__,__SREG__" "\n\t"    \
+    "cli" "\n\t"    \
+    "wdr" "\n\t"    \
+    "sts %0,%1" "\n\t"  \
+    "out __SREG__,__tmp_reg__" "\n\t"   \
+    "sts %0,%2" "\n\t" \
+    : /* no outputs */  \
+    : "M" (_SFR_MEM_ADDR(_WD_CONTROL_REG)), \
+    "r" (_BV(_WD_CHANGE_BIT) | _BV(WDE)), \
+    "r" ((uint8_t) ((value & 0x08 ? _WD_PS3_MASK : 0x00) | \
+        _BV(WDIE) | (value & 0x07)) ) \
+    : "r0"  \
+)
+
+#ifdef BLE_ENABLE
+#define ENABLE_SLEEP
+#define BATTERY_LEVEL
+#define LED_CONTROL
+// #define FAKE_BAT_LEVEL
+#define BAT_LEVEL_STEP_10
+#endif
+
+#include "grid_650.h"
+#include "mcp23018.h"
+
+// Optional override functions below.
+// You can leave any or all of these undefined.
+// These are only required if you want to perform custom actions.
+
+i2c_status_t mcp23018_status = 0x20;
+// bool ble_pwr_on = true;
+int batt_level = 50;
+ 
+
+enum LED_STAT {
+  NONE,
+  RED_BLINK,
+  WHITE_BREATHING,
+  WHITE_ON,  
+  BLE_ON,
+  ORANGE_ON  
+};
+
+#ifdef LED_CONTROL
+enum LED_STAT curr_stat = NONE;
+#endif
+
+void keyboard_pre_init_kb(void) {
+  // Call the keyboard pre init code.
+
+    //set orange led(PB7) for low
+    setPinOutput(B7);
+    writePinLow(B7);
+
+    //set red led(PB5) for low
+    setPinOutput(B5);
+    writePinLow(B5);
+    
+    //set solenoid
+    setPinOutput(C7);
+    writePinLow(C7);
+    
+    #ifdef BLE_ENABLE
+    //enable BLE_PWR(PD5)
+    setPinOutput(D5);
+    writePinHigh(D5);
+    
+    //enable BLE_OE(PD2)
+    setPinOutput(D2);
+    writePinLow(D2);
+
+    //disable MCU_DFU(PC6)
+    setPinOutput(C6);
+    writePinLow(C6);
+    #endif
+
+    //set VBUS DET(PD7) input
+    setPinInput(D7);
+    // setPinInputLow(D7);
+
+    //set CHG(PD3) input pullup
+    setPinInput(D3);
+    setPinInputHigh(D3);
+
+    #ifndef BLE_ENABLE
+    //set white LED high
+    setPinOutput(B6);
+    writePinHigh(B6);
+
+    //set orange led low
+    setPinOutput(B7);
+    writePinLow(B7);
+
+
+    #endif        
+
+    keyboard_pre_init_user();
+}
+
+
+#ifdef LED_CONTROL
+
+void set_status_led(enum LED_STAT led_stat,enum LED_STAT current_stat){
+    static bool red_is_on = false;
+    // xprintf("led stat = %d\n",led_stat);
+    // xprintf("bklt_level = %d\n",get_backlight_level());
+    // xprintf("is_breathing = %d\n",is_breathing());
```

Comment:
dprintf would probably be appropriate here, and wouldn't need commenting out. Specifically, dprintf only prints to console if debugging is enabled.

Reply:
use dprintf and uncommented